platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/DisambiguateArtifactTransformIntegrationTest.groovy

```diff
@@ -229,6 +229,7 @@ ${artifactTransform("TestTransform")}
 """
 
         when:
+        executer.expectDeprecationWarning("There are multiple distinct artifact transformation chains of the same length that would satisfy this request. This behavior has been deprecated. This will fail with an error in Gradle 9.0. ")
```

Comment:
Assume we were a user that got this warning after upgrading. 

How would I avoid it?

Does this test even simulate a real-world scenario?

Reply:
I want to make sure we test this scenario in the "passing" form. 

Whether that means copying the test or parameterizing this one